tools/plugins/preinstall/src/main/kotlin/net/corda/cli/plugins/preinstall/CheckKafka.kt

```
@@ -0,0 +1,151 @@
+        if (yaml.kafka.tls.enabled) {
+            if (yaml.kafka.sasl.enabled) {
+                props["security.protocol"] = "SASL_SSL"
+                props["sasl.mechanism"] = yaml.kafka.sasl.mechanism
+                if (yaml.kafka.sasl.mechanism == "PLAIN") {
+                    props["sasl.jaas.config"] =
+                        "org.apache.kafka.common.security.plain.PlainLoginModule required username=" +
+                                "\"$saslUsername\" password=\"$saslPassword\" ;"
+                } else {
+                    props["sasl.jaas.config"] =
+                        "org.apache.kafka.common.security.plain.ScramLoginModule required username=" +
+                                "\"$saslUsername\" password=\"$saslPassword\" ;"
+                }
+            } else {
+                props["security.protocol"] = "SSL"
+            }
+            yaml.kafka.tls.truststore?.valueFrom?.secretKeyRef?.name?.let {
+                props["ssl.truststore.location"] = truststoreLocation ?: run {
+                    log("If SSL is enabled, you must provide the location of the truststore file with -f.", ERROR)
+                    return null
+                }
+                props["ssl.truststore.type"] = yaml.kafka.tls.truststore.type.uppercase()
+                if (yaml.kafka.tls.truststore.password?.value != null
+                    || yaml.kafka.tls.truststore.password?.valueFrom?.secretKeyRef?.name != null
+                ) {
+                    props["ssl.truststore.password"] = truststorePassword
+                }
+            }
+        } else if (yaml.kafka.sasl.enabled) {
+            props["security.protocol"] = "SASL_PLAINTEXT"
+            props["sasl.mechanism"] = yaml.kafka.sasl.mechanism
+            props["sasl.jaas.config"] = "org.apache.kafka.common.security.scram.ScramLoginModule required username=" +
+                    "\"$saslUsername\" password=\"$saslPassword\" ;"
```

still need the option for `PLAIN` here

---

Is Plain ever used in the case where TLS is not enabled? I thought it was written this way to prevent this

---

Written where? The combination doesn't make great sense (sending credentials in plain across non-TLS connection) but I don't think Kafka prevents it.

---

The bootstrap template here https://github.com/corda/corda-runtime-os/blob/release/os/5.0/charts/corda-lib/templates/_bootstrap.tpl#L333

---

Either that was a response to a different comment or it's the wrong line!

---

Apologies, updated the link just now

---

yeah - I think that's just wrong!

---

I just assumed it was written to prevent the use of SASL Plain without TLS

---

if we did want to prevent it, it would be better to explicitly fail on that combination rather than just specifying the wrong login module...